include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -679,7 +678,127 @@ template <typename Lyt, typename TT>
 
     detail::is_operational_impl<Lyt, TT> p{lyt, spec, params};
 
-    return p.determine_operational_input_patterns();
+    std::set<uint64_t> input_patterns{};
```

Comment:
Is an `std::set` the best choice of data structure here? Couldn't an `std::vector` work?

Reply:
It is convenient because each entry is unique and automatically sorted, right?

Reply:
If those properties are more important, then gladly keep it as is. Just making sure it's not becoming a performance problem.

Reply:
Thank you! I agree, but I think it should be fine in this case.